Core/Configuration/JsonConfiguration.cs

```diff
@@ -0,0 +1,419 @@
+                    }
+                }
+
+                token = "";
+                return false;
+            }
+        }
+
+        public void SetAuthToken (string host, string token)
+        {
+            lock (_lock)
+            {
+                bool found = false;
+                foreach (AuthToken t in config.AuthTokens)
+                {
+                    if (t.Host == host)
+                    {
+                        found = true;
+                        t.Token = token;
+                    }
+                }
+
+                if (!found)
+                {
+                    config.AuthTokens.Add(new AuthToken
+                    {
+                        Host = host,
+                        Token = token
+                    });
+                }
+
+                SaveConfig();
+            }
+        }
+
+        // <summary>
+        // Save the JSON configuration file. Only call this while you own _lock.
+        // </summary>
+        private static void SaveConfig()
+        {
+            string json = JsonConvert.SerializeObject(config, Formatting.Indented);
+            File.WriteAllText(configFile, json);
+        }
+
+        // <summary>
+        // Load the JSON configuration file. This will replace the current state.
+        // Only call this while you own _lock.
+        //
+        // If the configuration file does not exist, this will create it and then
+        // try to populate it with values in the registry left from the old system.
+        // </summary>
+        private void LoadConfig()
+        {
+            try
+            {
+                string json = File.ReadAllText(configFile);
+                config = JsonConvert.DeserializeObject<Config>(json);
+
+                if (config == null)
+                {
+                    config = new Config();
+                }
+
+                if (config.KspInstances == null)
+                {
+                    config.KspInstances = new List<KspInstance>();
+
+                }
+
+                if (config.AuthTokens == null)
+                {
+                    config.AuthTokens = new List<AuthToken>();
+                }
+            }
+            catch (Exception ex) when (ex is FileNotFoundException || ex is DirectoryNotFoundException)
+            {
+                // This runs if the configuration does not exist. We will create a new configuration and
+                // try to migrate from the registry.
+                config = new Config();
+                config.KspInstances = new List<KspInstance>();
+                config.AuthTokens = new List<AuthToken>();
+
+                // Ensure the directory exists
+                new FileInfo(configFile).Directory.Create();
+
+                // Write the configuration to the disk
+                SaveConfig();
+
+#if !NETSTANDARD
+                // If we are not running on .NET Standard, try to migrate from the real registry
+                Migrate();
```

Comment:
Won't this overwrite the "new" config in the JSON file with the old values from the registry (which we can no longer edit) every time? I don't see any checks for whether the old or new settings are present.